### ISSUE TITLE
FluxPointEstimator fails above 100 TeV

### ISSUE THREAD

**Issue by robertazanin** [gammapy/gammapy#1638]
FluxPointEstimator fails estimating spectral points above 100 TeV. 

after 100 TeV, the energy is set back to 1 TeV

You can use as working example this notebook 

https://github.com/gammasky/cta-analyses/blob/master/dc-1-checks/hess_j1702/1D_spectral_analysis.ipynb

**Comment by cdeil** [comment]
cc @mealworm

**Comment by joleroi** [comment]
I can reproduce the issue with the following MWE
```
from gammapy.spectrum import SpectrumEnergyGroupMaker
from gammapy.utils.energy import EnergyBounds
import numpy as np
import astropy.units as u
energy_min = 0.1 * u.TeV
energy_max = 160 * u.TeV
ebounds = EnergyBounds.equal_log_spacing(energy_min, energy_max, 35)
class DummyObs(object): pass
obs = DummyObs()
obs.e_reco =  EnergyBounds.equal_log_spacing(10 * u.GeV, 100 * u.TeV, 72)
seq = SpectrumEnergyGroupMaker(obs=obs)
seq.compute_groups_fixed(ebounds=ebounds)
print(seq.groups)
```

which gives this output

```
SpectrumEnergyGroups:
energy_group_idx bin_idx_min bin_idx_max  bin_type   energy_min     energy_max  
                                                        TeV            TeV      
---------------- ----------- ----------- --------- -------------- --------------
               0           0          17 underflow           0.01            0.1
               1          18          19    normal            0.1 0.129154966501
               2          20          21    normal 0.129154966501  0.16681005372
               3          22          22    normal  0.16681005372 0.189573565241
               4          23          24    normal 0.189573565241 0.244843674682
               5          25          26    normal 0.244843674682 0.316227766017
               6          27          27    normal 0.316227766017  0.35938136638
               7          28          29    normal  0.35938136638 0.464158883361
               8          30          31    normal 0.464158883361 0.599484250319
               9          32          32    normal 0.599484250319 0.681292069058
              10          33          34    normal 0.681292069058 0.879922543569
              11          35          36    normal 0.879922543569  1.13646366639
              12          37          37    normal  1.13646366639  1.29154966501
              13          38          39    normal  1.29154966501   1.6681005372
              14          40          41    normal   1.6681005372  2.15443469003
              15          42          42    normal  2.15443469003  2.44843674682
              16          43          44    normal  2.44843674682  3.16227766017
              17          45          46    normal  3.16227766017  4.08423865267
              18          47          47    normal  4.08423865267  4.64158883361
              19          48          49    normal  4.64158883361  5.99484250319
              20          50          50    normal  5.99484250319  6.81292069058
              21          51          52    normal  6.81292069058  8.79922543569
              22          53          54    normal  8.79922543569  11.3646366639
              23          55          55    normal  11.3646366639  12.9154966501
              24          56          57    normal  12.9154966501   16.681005372
              25          58          59    normal   16.681005372  21.5443469003
              26          60          60    normal  21.5443469003  24.4843674682
              27          61          62    normal  24.4843674682  31.6227766017
              28          63          64    normal  31.6227766017  40.8423865267
              29          65          65    normal  40.8423865267  46.4158883361
              30          66          67    normal  46.4158883361  59.9484250319
              31          68          69    normal  59.9484250319  77.4263682681
              32          70          70    normal  77.4263682681  87.9922543569
              33          71          -1    normal  87.9922543569          100.0
              34           0          -1    normal           0.01          100.0
              35           0          71    normal           0.01          100.0


```

**Comment by cdeil** [comment]
Should be fixed by #1660 .
Closing this issue.

@robertazanin - Please try again with latest master.

**Comment by joleroi** [comment]
With the latest master I get
```
SpectrumEnergyGroups:
energy_group_idx bin_idx_min bin_idx_max  bin_type   energy_min     energy_max  
                                                        TeV            TeV      
---------------- ----------- ----------- --------- -------------- --------------
               0           0          17 underflow           0.01            0.1
               1          18          19    normal            0.1 0.129154966501
               2          20          20    normal 0.129154966501 0.146779926762
               3          21          22    normal 0.146779926762 0.189573565241
               4          23          24    normal 0.189573565241 0.244843674682
               5          25          25    normal 0.244843674682 0.278255940221
               6          26          27    normal 0.278255940221  0.35938136638
               7          28          29    normal  0.35938136638 0.464158883361
               8          30          30    normal 0.464158883361  0.52749970637
               9          31          32    normal  0.52749970637 0.681292069058
              10          33          33    normal 0.681292069058 0.774263682681
              11          34          35    normal 0.774263682681            1.0
              12          36          37    normal            1.0  1.29154966501
              13          38          38    normal  1.29154966501  1.46779926762
              14          39          40    normal  1.46779926762  1.89573565241
              15          41          42    normal  1.89573565241  2.44843674682
              16          43          43    normal  2.44843674682  2.78255940221
              17          44          45    normal  2.78255940221   3.5938136638
              18          46          47    normal   3.5938136638  4.64158883361
              19          48          48    normal  4.64158883361   5.2749970637
              20          49          50    normal   5.2749970637  6.81292069058
              21          51          52    normal  6.81292069058  8.79922543569
              22          53          53    normal  8.79922543569           10.0
              23          54          55    normal           10.0  12.9154966501
              24          56          57    normal  12.9154966501   16.681005372
              25          58          58    normal   16.681005372  18.9573565241
              26          59          60    normal  18.9573565241  24.4843674682
              27          61          61    normal  24.4843674682  27.8255940221
              28          62          63    normal  27.8255940221   35.938136638
              29          64          65    normal   35.938136638  46.4158883361
              30          66          66    normal  46.4158883361   52.749970637
              31          67          68    normal   52.749970637  68.1292069058
              32          69          70    normal  68.1292069058  87.9922543569
              33          71          71    normal  87.9922543569          100.0
```

as expected :fireworks:
